File: app/src/main/java/de/werkelmann/rekote/client/tasks/CallUrlTask.java
package de.werkelmann.rekote.client.tasks;

import android.os.AsyncTask;

import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URL;

public class CallUrlTask extends AsyncTask<URL, Void, Boolean> {
    @Override
    protected Boolean doInBackground(URL... params) {
        try {
            URL url = params[0];
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
            connection.setConnectTimeout(5000);
            connection.connect();

            return (connection.getResponseCode() == 200);
        } catch (IOException e) {
            return false;
        }
    }
}

File: app/src/main/kotlin/de/werkelmann/rekote/client/tasks/CallUrlTask.kt
package de.werkelmann.rekote.client.tasks

import android.os.AsyncTask

import java.io.IOException
import java.net.HttpURLConnection
import java.net.URL

class CallUrlTask : AsyncTask<URL, Void, Boolean>() {
    override fun doInBackground(vararg params: URL): Boolean? {
        try {
            val url = params[0]
            val connection = url.openConnection() as HttpURLConnection
            connection.connectTimeout = 5000
            connection.connect()
            return connection.responseCode == 200
        } catch (e: IOException) {
            return false
        }

    }
}
